lms/models/region.py

```diff
@@ -0,0 +1,44 @@
+from dataclasses import dataclass
+
+from pyramid.request import Request
+
+
+@dataclass(frozen=True)
+class Region:
+    """Codes and details related to a deployment region."""
+
+    code: str
+    """Region code (e.g. ISO 3166 alpha-2 country code)."""
+
+    name: str
+    """Human readable description of the region."""
+
+    authority: str
+    """Associated Hypothesis authority."""
+
+
+class Regions:
+    """A collection of all the regions as an enum like object."""
+
+    # As region objects are frozen, these can be directly compared like enums
+    US = Region(code="us", name="Worldwide (U.S.)", authority="lms.hypothes.is")
+    CA = Region(code="ca", name="Canada", authority="lms.ca.hypothes.is")
+    ALL = US, CA
+
+    _AUTHORITY_MAP = {region.authority: region for region in ALL}
+
+    @classmethod
+    def from_authority(cls, request: Request) -> Region:
+        """
+        Get a region object based on a request.
+
+        :raises ValueError: If no valid region can be found.
+        """
+        authority = request.registry.settings["h_authority"]
+
+        try:
+            return cls._AUTHORITY_MAP[authority]
+        except KeyError as err:
+            raise ValueError(
+                f"Cannot find a region for the authority '{authority}'"
+            ) from err
```

Comment:
This part isn't in use yet, so perhaps it shouldn't be in this PR? But I wanted to show how we might go about getting this.

This is setting up for either:

 * Being called from a service
 * Being added to the request as a request method (`request.region`)
 
The second is sort of "nice" in that it's very short, but it sucks because it pollutes the global namespace for a specific purpose. We don't care about the region in a large number of places (any yet).

Reply:
Not sure if the h_authority 1 to 1 relationship to region is the way to go (will that always be true?)

We could add a new `region_code` setting but it's true that new variables have a bit of operational complexity and if now 1 authority 1 region, why not use it :shrug: